keps/sig-node/3063-dynamic-resource-allocation/README.md

```diff
@@ -2024,22 +2024,40 @@ message NodePrepareResourceResponse {
 }
 ```
 
-CRI protocol MUST be extended for this purpose, e.g. list of CDI
-device ids should be added to the CRI Device structure:
+CRI protocol MUST be extended for this purpose:
 
+ * CDIDevice structure should be added to the CRI specification
 ```protobuf
-// Device specifies a host device to mount into a container.
-message Device {
-    ...
-    string permissions = 3;
-    // Set of fully qualified CDI device names in the following
+// CDIDevice specifies a CDI device information.
+message CDIDevice {
+    // Fully qualified CDI device name in the following
     // format: <kind>=<name>,
     //  where
     //    kind (string) is a device vendor identifier
     //    name (string) is a device name
```

Comment:
Are we still following this format? Because then we could have `kind` and `name` as separate fields.

Reply:
that might make sense... if whole CDIDevice struct is repeated, inside of it having separate kind/name might be ok.
On other hand NodePrepareResourceResponse returns just one string. Having here two fields means that we would need to parse NodePrepareResourceResponse inside kubelet to make it transformed to pass inside CRI.
So, we either keep inside CDIDevice simple string (maybe "name" is not good field name) or we would need to adjust NodePrepareResourceResponse to match return value format there and in CRI, to avoid complexity inside kubelet to parse/transform data types.

Reply:
@bart0sh: so which of the two possible non-parsing approaches do you prefer?

- pass a single string around
- split the fields in both CRI and the NodePrepareResource gRPC call

Reply:
@pohly I'd prefer to pass a single string around. This is what we already have, so we don't have to modify existing DRA APIs.

Reply:
Can we add an example of what it will look like?

Reply:
nvm, I see it below.

Reply:
Do we expect to pass any other information in the future here?

Reply:
@mrunalp We don't expect any particular additions yet. However we decided to have a separate structure `CDIDevice`  just in case we need to add new fields to it in the future.

Reply:
@bart0sh okay! I think it will be cleaner to separate the `kind` and `name` in the CRI struct.

Reply:
@mrunalp we already discussed this and came to the opposite conclusion. See @klueska's comment below for more details: https://github.com/kubernetes/enhancements/pull/3731#issuecomment-1398078642

@kad @klueska thoughts?

Reply:
I agree with @klueska  and state that in comment to @saschagrunert, splitting here simple string into two fields add unneded data conversion code in kubelet, which we are trying to avoid. To kubelet it should be treated as "unique identifier" without trying to figure out from which parts this identifier might consist of.

Reply:
as a side note, we can at all forget that it is a CDI related string. we can have `ContainerResourceUUID` kind of field and potentially `PodResourceUUID` field to remove assumptions that those are CDI specific.
(I'm bad with naming, so not proposing the names as such, but just to demonstrate the idea)

Reply:
But then the producer and user of those fields would still have to agree that they are using them for CDI strings in a particular format. We should have that convention documented in the API, therefore the current proposal makes sense to me.

Reply:
It looks like specifying format of the CDI device name creates confusions. I'm considering to removing that comment from this KEP as it doesn't make sense to know device name format at any point of the DRA data path (plugin->Kubelet->CRI->Runtime).

PS: Nobody proposes to pass e.g. file path as a directory and file name or [FQDN](https://en.wikipedia.org/wiki/Fully_qualified_domain_name) as a list of domain labels.
Fully Qualified CDI Device Name is a similar term in my point of view and should be passed as is without parsing back and force on every intermediate level.

Reply:
We don't need to specify what the format of that string is, but we do need a definition somewhere of what is and what isn't valid here.

Reply:
I agree. It should be in CDI docs I believe. 
Here is what I found there:
1. [In the CDI device model containers request access to a device using fully qualified device names, qualified names for short, consisting of a vendor identifier, a device class and a device name or identifier. These pieces of information together uniquely identify a device among all device vendors, classes and device instances.](https://github.com/container-orchestrated-devices/container-device-interface/blob/main/pkg/cdi/doc.go#L20-L24)
2. [// The syntax for a qualified device names is
//   `<vendor>/<class>=<name>`.
// A valid vendor name may contain the following runes:
//   'A'-'Z', 'a'-'z', '0'-'9', '.', '-', '_'.
// A valid class name may contain the following runes:
//   'A'-'Z', 'a'-'z', '0'-'9', '-', '_'.
// A valid device name may containe the following runes:
//   'A'-'Z', 'a'-'z', '0'-'9', '-', '_', '.', ':'](https://github.com/container-orchestrated-devices/container-device-interface/blob/main/pkg/cdi/qualified-device.go#L24-L32)
 3. [ `name` (string, REQUIRED), name of the device, can be used to refer to it when requesting a device.
       Beginning and ending with an alphanumeric character ([a-z0-9A-Z]) with dashes (-), underscores (\_), dots (.), and alphanumerics between.
       e.g: `docker/podman run --device foo ...`
      Entries in the array MUST use the same schema as the entry for the `name` field](https://github.com/container-orchestrated-devices/container-device-interface/blob/main/SPEC.md?plain=1#L165-L168)

It probably makes sense to update `name` definition in the SPEC.md a little bit, but even what I've found explained the term quite clearly in my opinion.

Reply:
@klueska @kad @pohly @mrunalp  I've removed confusing format details. PTAL.